src/aliceVision/image/dcp.hpp

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <map>
+#include <vector>
+#include <array>
+#include <memory>
+#include <string>
+
+#include <OpenImageIO/imagebuf.h>
+
+#include "curves.hpp"
+
+namespace alicevision
+{
+    namespace image
+    {
+        // Color space conversion to/from XYZ; color spaces adapted to D50 using Bradford transform
+        // Source : Bruce Lindbloom's website : http://www.brucelindbloom.com/
+        constexpr double xyz_sRGB[3][3] = {
+            {0.4360747,  0.3850649, 0.1430804},
+            {0.2225045,  0.7168786,  0.0606169},
+            {0.0139322,  0.0971045,  0.7141733}
+        };      
+        constexpr double sRGB_xyz[3][3] = {
+            {3.1338561, -1.6168667, -0.4906146},
+            { -0.9787684,  1.9161415,  0.0334540},
+            {0.0719453, -0.2289914,  1.4052427}
+        };
+        
+        constexpr double xyz_prophoto[3][3] = {
+            {0.7976749,  0.1351917,  0.0313534},
+            {0.2880402,  0.7118741,  0.0000857},
+            {0.0000000,  0.0000000,  0.8252100}
+        };        
+        constexpr double prophoto_xyz[3][3] = {
+            {1.3459433, -0.2556075, -0.0511118},
+            { -0.5445989,  1.5081673,  0.0205351},
+            {0.0000000,  0.0000000,  1.2118128}
+        };
+
+        constexpr double xyz_AdobeRGB[3][3] = {
+            {0.6097559,  0.2052401,  0.1492240},
+            {0.3111242,  0.6256560,  0.0632197},
+            {0.0194811,  0.0608902,  0.7448387}
+        };
+        constexpr double AdobeRGB_xyz[3][3] = {
+            { 1.9624274, -0.6105343, -0.3413404},
+            {-0.9787684,  1.9161415,  0.0334540},
+            { 0.0286869, -0.1406752,  1.3487655}
+        };
+
+        typedef struct DCPProfileApplyParams {
+            bool apply_hue_shift;
+            bool apply_baseline_exposure_offset;
+            bool use_tone_curve;
+            bool apply_look_table;
+            std::string working_space;
+
+            DCPProfileApplyParams()
+            {
+                apply_hue_shift = true;
+                apply_baseline_exposure_offset = true;
+                use_tone_curve = true;
+                apply_look_table = true;
+                working_space = "sRGB";
+            }
+
+        } DCPProfileApplyParams;
```

Comment:
typedef not needed in c++
```suggestion
       struct DCPProfileApplyParams {
            bool apply_hue_shift;
            bool apply_baseline_exposure_offset;
            bool use_tone_curve;
            bool apply_look_table;
            std::string working_space;

            DCPProfileApplyParams()
            {
                apply_hue_shift = true;
                apply_baseline_exposure_offset = true;
                use_tone_curve = true;
                apply_look_table = true;
                working_space = "sRGB";
            }

        };
```